src/components/KTreeList/KTreeDraggable.vue

```diff
@@ -110,61 +104,25 @@ export const getMaximumDepth = ({ children = [] }): number => {
 </script>
 
 <script setup lang="ts">
-import { nextTick, useTemplateRef } from 'vue'
-
-const props = defineProps({
-  items: {
-    type: Array as PropType<TreeListItem[]>,
-    required: true,
-    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
-  },
-  disableDrag: {
-    type: Boolean,
-    default: false,
-  },
-  filter: {
-    type: [String, Function] as PropType<string | (() => string)>,
-    default: () => '',
-  },
-  maxDepth: {
-    type: Number,
-    default: 3,
-  },
-  /**
-   * These are internal props used to track info
-   * of nested children.
-   */
-  level: {
-    type: Number,
-    default: 0,
-  },
-  parentId: {
-    type: String,
-    default: '',
-  },
-  hideIcons: {
-    type: Boolean,
-    default: false,
-  },
-  group: {
-    type: String,
-    default: 'k-tree-list',
-  },
-  collapsible: {
-    type: Boolean,
-    default: false,
-  },
-  initialCollapseAll: {
-    type: Boolean,
-    default: false,
-  },
-})
-
-const emit = defineEmits<{
-  (event: 'change', data: TreeListChangeEvent): void
-  (event: 'child-change', data: TreeListChildChangeEvent): void
-  (event: 'selected', item: TreeListItem): void
-}>()
+import { computed, ref, watch, onMounted, nextTick, useTemplateRef } from 'vue'
+import { VueDraggableNext } from 'vue-draggable-next'
+import KTreeItem from '@/components/KTreeList/KTreeItem.vue'
+import type { TreeDraggableProps, TreeDraggableEmits, TreeListItem } from '@/types'
+
+const {
+  items,
+  disableDrag,
+  filter,
+  maxDepth,
+  level = 0,
+  parentId,
+  hideIcons,
+  group,
+  collapsible,
+  initialCollapseAll,
```

Comment:
Removed default values from the private component’s props, as they’re already defined in the public parent. This avoids redundancy and keeps a single source of truth.